### PR TITLE
CLIO-23779 Refactor code layout.

### PR DESCRIPTION
**Pull request by kyledemeule**

### CODE REVIEW

File: src/Zap.coffee [hunk 1]
```diff
@@ -1,12 +1,61 @@
...
+  url = Zap.build_api_url(bundle.auth_fields.domain, resource_url)
+  JSON.parse z.request(Zap.build_request(bundle, url, "POST", data)).content
+
+################################################################################
+#### Helper methods
+################################################################################
+
+Zap.typeIsArray = Array.isArray || ( value ) -> return {}.toString.call( value ) is '[object Array]'
+Zap.valueExists = (value) -> value not in ["", null, undefined]
+Zap.valueMissing = (value) -> value in ["", null, undefined]
```
Comment (alecclarke) [inline src/Zap.coffee]: typeIsArray, valueExists, valueMissing are the only camelCase functions, should they be changed to snake_case?

File: src/Zap.coffee [hunk 2]
```diff
@@ -1,12 +1,61 @@
...
+
+Zap.make_get_request =  (bundle, resource_url) ->
+  url = Zap.build_api_url(bundle.auth_fields.domain, resource_url)
+  content = z.request(Zap.build_request(bundle, url, "GET", null)).content
+  JSON.parse content
+
+Zap.make_post_request = (bundle, resource_url, data) ->
+  url = Zap.build_api_url(bundle.auth_fields.domain, resource_url)
+  JSON.parse z.request(Zap.build_request(bundle, url, "POST", data)).content
+
```
Comment (alecclarke) [inline src/Zap.coffee]: Maybe change Zap.make_post_request to:

Zap.make_post_request = (bundle, resource_url, data) ->
 url = Zap.build_api_url(bundle.auth_fields.domain, resource_url) 
 content = z.request(Zap.build_request(bundle, url, "POST", data)).content
 JSON.parse content

to be consistent with make_get_request, or change  make_get_request to be consistent with make_post_request.

File: src/activity.coffee [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    data.matter = Zap.transform_nested_attributes(object.matter)
+    data.bill = Zap.transform_nested_attributes(object.bill)
+    array.push data
+  array
+
+################################################################################
+#### Helper methods
+################################################################################
+
+Zap.create_activity_pre_write = (bundle, activity_type) ->
```
Comment (alecclarke) [inline src/activity.coffee]: I feel like ...pre_write, and the other Zapier hook prefixes should only be used for the actual hook methods, so maybe change Zap.create_activity_pre_write to Zap.create_activity, thoughts?

Comment (alecclarke) [inline src/activity.coffee]: Zapier hook suffixes*

File: src/contact.coffee [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  Zap.transform_custom_action_fields(bundle, false)
+
+Zap.create_company_post_custom_action_fields = (bundle) ->
+  Zap.transform_custom_action_fields(bundle, false)
+
+################################################################################
+#### Helper methods
+################################################################################
+
+Zap.create_contact_pre_write = (bundle, contact_type) ->
```
Comment (alecclarke) [inline src/contact.coffee]: Same "Zapier suffix" method naming question as before.

File: src/matter.coffee [hunk 5]
```diff
@@ -98,3 +66,118 @@ Zap.create_matter_pre_write = (bundle) ->
   
   bundle.request.data = JSON.stringify({"matter": data})
   bundle.request
+
+Zap.create_person_and_matter_pre_write = (bundle) ->
```
Comment (alecclarke) [inline src/matter.coffee]: Same as before.

File: src/matter.coffee [hunk 6]
```diff
@@ -98,3 +66,118 @@ Zap.create_matter_pre_write = (bundle) ->
...
+    data.originating_attorney = Zap.transform_nested_attributes(object.originating_attorney)
+    data.practice_area = Zap.transform_nested_attributes(object.practice_area)
+    data.custom_fields = Zap.transform_custom_fields(bundle, object, "Matter")
+    array.push data
+  array
+
+Zap.create_matter_post_custom_action_fields = (bundle) ->
+  Zap.transform_custom_action_fields(bundle, false)
+
+Zap.create_person_and_matter_post_custom_action_fields = (bundle) ->
```
Comment (alecclarke) [inline src/matter.coffee]: Again, not sure if create_person_and_matter stuff is actually used anywhere.

File: src/note.coffee [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
   Zap.create_note_pre_write(bundle, "Contact")
+
 Zap.create_matter_note_pre_write = (bundle) ->
   Zap.create_note_pre_write(bundle, "Matter")
 
+################################################################################
+#### Helper methods
+################################################################################
+
 Zap.create_note_pre_write = (bundle, regarding_type) ->
```
Comment (alecclarke) [inline src/note.coffee]: Same "Zapier suffix" method naming question as before.

File: src/Zap.coffee (hunk 2 above)
Comment (kyledemeule) [inline src/Zap.coffee]: You can wrap code in ``` to make it look like code, i.e.
```
Zap.make_post_request = (bundle, resource_url, data) ->
  url = Zap.build_api_url(bundle.auth_fields.domain, resource_url) 
  content = z.request(Zap.build_request(bundle, url, "POST", data)).content
  JSON.parse content
```

File: src/activity.coffee (hunk 3 above)
Comment (kyledemeule) [inline src/activity.coffee]: create_activity is a deprecated action... Zapier doesn't currently have starts on hidden actions so Im not sure if anyone is using it. I like the idea but im just going to leave this one alone.

File: src/contact.coffee (hunk 4 above)
Comment (kyledemeule) [inline src/contact.coffee]: same answer.

File: src/matter.coffee (hunk 6 above)
Comment (kyledemeule) [inline src/matter.coffee]: Another deprecated action

File: src/activity.coffee (hunk 3 above)
Comment (alecclarke) [inline src/activity.coffee]: Ah right, forgot that they were still in use, thanks.

**Pull request merged**